Lib/fontTools/ttLib/ttFont.py

```diff
@@ -666,7 +666,12 @@ def getGlyphNameMany(self, lst):
     def getGlyphID(self, glyphName):
         """Returns the ID of the glyph with the given name."""
         try:
-            return self.getReverseGlyphMap()[glyphName]
+            gid = self.getReverseGlyphMap()[glyphName]
+            if self.getGlyphName(gid) == glyphName:
+                return gid
+            self._buildReverseGlyphOrderDict()
```

Comment:
@behdad — thank you for the quick turn around. With this change as is, it still fails for me, but if I move 672 before 669 then it works fine, i.e., rebuild first.

Reply:
fails how exactly? have you got a test case that reproduces this?
also it isn't clear to me what you mean by 'move 672 before 669'

Reply:
@xadxura I just saw your other comment now

> no glyph ID was returned for cases where I appended new glyphs to the font. These return None

I think the latest @behdad's patch does not return a None gid any more. Are you sure you tested with the latest latest commit? Try to git fetch origin && git reset --hard origin/getGlyphID (where `origin` in my case is the upstream remote)